File: app/src/main/java/ch/unstable/ost/views/NoAnimationStrategy.java
package ch.unstable.ost.views;

import android.support.annotation.MainThread;
import android.support.annotation.NonNull;
import android.view.View;

/**
 * Animation strategy that doesn't animate anything.
 */
public class NoAnimationStrategy implements AnimationStrategy {

    @MainThread
    @Override
    public void initHiddenView(@NonNull View view) {
        view.setVisibility(View.GONE);
    }

    @MainThread
    @Override
    public void initShownView(@NonNull View view) {
        view.setVisibility(View.VISIBLE);
    }

    @MainThread
    @Override
    public void showView(@NonNull View view) {
        view.setVisibility(View.VISIBLE);
    }

    @MainThread
    @Override
    public void hideView(@NonNull View view) {
        view.setVisibility(View.GONE);
    }

    @MainThread
    @Override
    public void crossFadeViews(@NonNull View viewToHide, @NonNull View viewToShow) {
        hideView(viewToHide);
        showView(viewToShow);
    }
}

File: app/src/main/java/ch/unstable/ost/views/NoAnimationStrategy.kt
package ch.unstable.ost.views

import android.view.View
import androidx.annotation.MainThread

/**
 * Animation strategy that doesn't animate anything.
 */
class NoAnimationStrategy : AnimationStrategy {
    @MainThread
    override fun initHiddenView(view: View) {
        view.visibility = View.GONE
    }

    @MainThread
    override fun initShownView(view: View) {
        view.visibility = View.VISIBLE
    }

    @MainThread
    override fun showView(view: View) {
        view.visibility = View.VISIBLE
    }

    @MainThread
    override fun hideView(view: View) {
        view.visibility = View.GONE
    }

    @MainThread
    override fun crossFadeViews(hideView: View, viewToShow: View) {
        hideView(hideView)
        showView(viewToShow)
    }
}